src/bin/common/solving_loop.ml

```diff
@@ -50,8 +50,50 @@ let empty_solver_ctx = {
   global = [];
 }
 
-let unsupported_opt () =
-  Printer.print_std "unsupported"
+let custom_error
+    (type a)
+    ?(default : a option)
+    err_code
+    (msg : ('a, Format.formatter, unit, unit, unit, a) format6)
```

Comment:
I understand the reason for it, but I feel like this type annotation is overly complicated for little benefit.

What about something like:

```ocaml
let recoverable_error ?(code = 1) msg =
  Format.kasprintf (fun msg ->
     let () =
      if msg <> "" then
        match Options.get_output_format () with
        | Smtlib2 -> Printer.print_smtlib_err "%s" msg
        | _ -> Printer.print_err "%s" msg
    in
    if Options.get_exit_on_error () then exit code)

let fatal_error ?code msg =
  recoverable_error ?code msg;
  exit (Option.value ~default:1 code)
```

The use of `fatal_error` vs `recoverable_error` is clearer than `error` vs `error ~default:v` IMO.